src/solver/mod.rs

```diff
@@ -732,8 +684,25 @@ impl<VS: VersionSet, N: PackageName + Display, D: DependencyProvider<VS, N>> Sol
     /// way, the clause can check whether the literal that is using the solvable has become false, in
     /// which case it picks a new solvable to watch (if available) or triggers an assignment.
     fn propagate(&mut self, level: u32) -> Result<(), (SolvableId, bool, ClauseId)> {
-        // Learnt assertions (assertions are clauses that consist of a single literal, and therefore
-        // do not have watches)
+        // Negative assertions derived from other rules (assertions are clauses that consist of a
+        // single literal, and therefore do not have watches)
+        for &(solvable_id, clause_id) in &self.negative_assertions {
```

Comment:
Propagation is done a lot, this now runs for all `negative_assertions` every time `propagate` is called. Would it make sense to only do this for a subset of the negative assertions?

Reply:
Should we rerun the benchmarks in `rattler`?

Reply:
I think running the benchmarks makes sense. It's been a long time since I wrote this code, but I was under the impression that this part of propagation is not run _that_ often (only after an arbitrary decision has been made). Also, it's not clear to me how to select the right subset of negative assertions without introducing complexity. With a bit of luck, the perf impact will be negligible and we can get away with it.

Reply:
(For future reference: the benchmarks showed no significant difference / a slight improvement)